Split modules between NodeJS and browser

emizzle opened the pull request:
First version of a swarm js library for interacting with a swarm gateway

The library is transpiled by babel to work with Node and the browser. The `uploadDirectory` function is only available in Node (due to the `fs` requirement).

Still left to do:
- Integrate with ENS so uploaded content can be accessed via an ENS name instead of a hash
- Modify `uploadDirectory` function so that the `defaultPath` can be specified (currently it doesn't work)
- unit tests

Related to https://github.com/embark-framework/embark/pull/802.

Commit:
Split module between nodejs and browser

Somehow when embark webpacks the dapp, it is still seeing the ‘fs’ object, which of course it doesn’t know about, so it is complaining about a missing module. Any help would be greatly appreciated.

Commit:
Split modules working

Fixed issues with split modules not working

Updated readme

Clean up

Commit:
Update README.md

Commit:
more readme updates

Commit:
Re-add babel script (removed in rebase)

Commit:
Update files for tarball

jrainville commented on src/index.js:
```diff
@@ -1,34 +1,23 @@
-    });
+      if (!item.stats.isDirectory()) this.push(item)
+      next()
+    })
```
Why remove the `;`?

emizzle replied:
Good catch, this was a rebase problem. Will fix.

jrainville commented on src/shared.js:
```diff
@@ -0,0 +1,82 @@
+    } else if (this.opts.mode === 'http') {
+      this.gateway = 'http://swarm-gateways.net'
+    } else {
+      this.gateway = 'https://swarm-gateways.net'
```
You have swarm-gateway twice. Maybe you can remove the `else if` then?

emizzle replied:
Haha I knew it would trigger you 😜 I'll extract the `swarm-gateways.net`.

alaibe commented on swarmjs.min.js:
```diff
@@ -0,0 +1 @@
+!function(t,e){"object"==typeof exports&&"object"==typeof module?module.exports=e():"function"==typeof define&&define.amd?define("SwarmJS",[],e):"object"==typeof exports?exports.SwarmJS=e():t.SwarmJS=e()}("undefined"!=typeof self?self:this,function(){return function(t){var e={};function n(r){if(e[r])return e[r].exports;var o=e[r]={i:r,l:!1,exports:{}};return t[r].call(o.exports,o,o.exports,n),o.l ...
```
Is this file supposed to be version controlled?

emizzle replied:
Good point, I must've added it to the .gitignore after it was already added or possibly got readded in a rebase. Will remove from the index.

alaibe commented on webpack.config.js:
```diff
@@ -0,0 +1,26 @@
+const path = require('path');
+
+const standalone = {
+    entry: path.join(__dirname, 'dist/browser', 'browser.js'),
```
maybe keep 2 spaces indentation everywhere?

emizzle replied:
100%, good catch

Commit:
Constructor refactor, clean up

 Linting, remove swarmjs.min, format spaces

Refactored constructor of SwarmJS to remove `mode` option. By default, if no `gateway` option is passed in, the default gateway will be `https://swarm-gateways.net`.  Removed the `else if` and only list `swarm-gateways.net` once.

Commit:
Rename to SwarmAPI

Commit:
Rename to swarm-api, swarmapi CLI updated

Renamed library to swarm-api to appease npm

swarmapi NodeJS CLI bin updated to use all of the newest functions in the library

README updated with API doc

Commit:
version bump to 0.1.0
